src/expr/src/scalar/func.rs

```diff
@@ -3397,6 +3397,10 @@ trait LazyUnaryFunc {
     /// is used by the optimizer when a guarantee can be made that a collection
     /// with unique items will stay unique when mapped by this function.
     ///
+    /// Note that error results are not covered: Even with `preserves_uniqueness = true`, it can
+    /// happen that two different inputs produce the same error result. (e.g., in case of a
+    /// narrowing cast)
```

Comment:
Maybe an example that illustrates this will be good here.

Reply:
Ok, I'll extend the comment. For example, casting the integer 1000000 to smallint yields the same error result as casting the integer 2000000 to smallint.